pkg/apis/componentconfig/v1alpha1/types.go

```diff
@@ -108,9 +108,13 @@ type KubeSchedulerConfiguration struct {
 	FailureDomains string `json:"failureDomains"`
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
 // LeaderElectionConfiguration defines the configuration of leader election
 // clients for components that can run with leader election enabled.
 type LeaderElectionConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
```

Comment:
Why does this need type info? It's not a top level type, right?

Reply:
@liggitt some of the tests fail without it

Reply:
unless this is serialized in standalone way (which I'm not aware of), this should not have type info. tests that fail without that should be fixed.

Reply:
ack removing it